lib/iris/tests/integration/netcdf/test__chunk_control.py

```diff
@@ -0,0 +1,157 @@
+    with CHUNK_CONTROL.set(model_level_number=2):
+        cube = iris.load_cube(tmp_filepath, create_cube[0])
+    assert cube.shape == (3, 4, 5, 6)
+    assert cube.lazy_data().chunksize == (3, 2, 5, 6)
+
+    sigma = cube.coord("sigma")
+    assert sigma.shape == (4,)
+    assert sigma.lazy_points().chunksize == (2,)
+    assert sigma.lazy_bounds().chunksize == (2, 2)
+
+
+def test_control_sigma_only(tmp_filepath, create_cube):
+    with CHUNK_CONTROL.set(create_cube[1], model_level_number=2):
+        cube = iris.load_cube(tmp_filepath, create_cube[0])
+    assert cube.shape == (3, 4, 5, 6)
+    assert cube.lazy_data().chunksize == (3, 4, 5, 6)
+
+    sigma = cube.coord("sigma")
+    assert sigma.shape == (4,)
+    assert sigma.lazy_points().chunksize == (2,)
+    assert sigma.lazy_bounds().chunksize == (4, 2)
+
+
+def test_control_cube_var(tmp_filepath, create_cube):
+    with CHUNK_CONTROL.set(create_cube[0], model_level_number=2):
+        cube = iris.load_cube(tmp_filepath, create_cube[0])
+    assert cube.shape == (3, 4, 5, 6)
+    assert cube.lazy_data().chunksize == (3, 2, 5, 6)
+
+    sigma = cube.coord("sigma")
+    assert sigma.shape == (4,)
+    assert sigma.lazy_points().chunksize == (2,)
+    assert sigma.lazy_bounds().chunksize == (2, 2)
+
+
+def test_control_multiple(tmp_filepath, create_cube):
+    with CHUNK_CONTROL.set(
+        create_cube[0], model_level_number=2
+    ), CHUNK_CONTROL.set(create_cube[1], model_level_number=3):
+        cube = iris.load_cube(tmp_filepath, create_cube[0])
+    assert cube.shape == (3, 4, 5, 6)
+    assert cube.lazy_data().chunksize == (3, 2, 5, 6)
+
+    sigma = cube.coord("sigma")
+    assert sigma.shape == (4,)
+    assert sigma.lazy_points().chunksize == (3,)
+    assert sigma.lazy_bounds().chunksize == (2, 2)
+
+
+def test_neg_one(tmp_filepath, create_cube):
+    with CHUNK_CONTROL.set(model_level_number=-1):
+        cube = iris.load_cube(tmp_filepath, create_cube[0])
+    assert cube.shape == (3, 4, 5, 6)
+    assert cube.lazy_data().chunksize == (3, 4, 5, 6)
+
+    sigma = cube.coord("sigma")
+    assert sigma.shape == (4,)
+    assert sigma.lazy_points().chunksize == (4,)
+    assert sigma.lazy_bounds().chunksize == (4, 2)
+
+
+def test_from_file(tmp_filepath, create_file_cube):
+    with CHUNK_CONTROL.from_file():
+        cube = iris.load_cube(tmp_filepath)
+    assert cube.shape == (2, 3, 4)
+    assert cube.lazy_data().chunksize == (1, 3, 4)
+
+
+def test_as_dask(tmp_filepath, create_cube):
+    message = "Mock called, rest of test unneeded"
+    with patch("iris.fileformats.netcdf.loader.as_lazy_data") as optimum:
+        optimum.side_effect = RuntimeError(message)
+        with CHUNK_CONTROL.as_dask():
+            try:
+                iris.load_cube(tmp_filepath, create_cube[0])
+            except RuntimeError as e:
+                if str(e) == message:
+                    pass
+                else:
+                    raise e
```

Comment:
This is the alternative, if you are interested:

```suggestion
        class AsDaskError(Exception):
            # Allows calling the Mock without needing defined Mock behaviour.
            pass
        
        optimum.side_effect = AsDaskError(message)
        with CHUNK_CONTROL.as_dask():
            try:
                iris.load_cube(tmp_filepath, create_cube[0])
            except AsDaskErroras:
                pass
```

Reply:
I don't think there's much merit in one over the other, and since I already have one functioning, I'll probably leave as is.